src/workerd/api/http.c++

```diff
@@ -1867,6 +1868,36 @@ jsg::Promise<jsg::Ref<Response>> Fetcher::fetch(
   return fetchImpl(js, JSG_THIS, kj::mv(requestOrUrl), kj::mv(requestInit));
 }
 
+kj::Maybe<Fetcher::RpcFunction> Fetcher::getRpcMethod(jsg::Lock& js, kj::StringPtr name) {
+  // This is like JsRpcCapability::getRpcMethod(), but we also initiate a whole new JS RPC session
+  // each time the method is called.
+  return RpcFunction(JSG_VISITABLE_LAMBDA(
+      (methodName = kj::str(name), self = JSG_THIS),
+      (self),
+      (jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) -> jsg::Promise<jsg::Value> {
+    // Same as JsRpcCapability::getRpcMethod(), we want to prevent calling on the wrong `this`.
+    JSG_REQUIRE(args.This() == KJ_ASSERT_NONNULL(self.tryGetHandle(js)), TypeError,
+        "Method '", methodName, "' called with wrong 'this'.");
```

Comment:
Nit: `Invalid invocation` for consistency or just say screw it and go with a better error message? Either way is fine but let's make sure the decision is intentional.